api/api.yaml

```diff
@@ -0,0 +1,299 @@
+swagger: "2.0"
+info:
+  description: "This is a explorer service. It can be used to create, view, modify and delete explorers. It is also used to login and logout of the system."
+  version: "1.0.0"
+  title: "Explorer Service API"
+  contact:
+    email: "prashanth.ramakrishnan@gmail.com"
+  license:
+    name: "MIT License"
+    url: "https://opensource.org/licenses/MIT"
+host: "explorerservice.swagger.io"
+basePath: "/v2"
+tags:
+- name: "explorer"
+  description: "API to perform different operations on explorer"
+- name: "team"
+  description: "API to perform different operations on team"
+schemes:
+- "https"
+- "http"
+paths:
+  /explorer:
+    post:
+      tags:
+      - "explorer"
+      summary: "Create explorer"
+      description: "This can only be done by the logged in explorer."
+      operationId: "createExplorer"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Created explorer object"
+        required: true
+        schema:
+          $ref: "#/definitions/Explorer"
+      responses:
+        201:
+          description: "successful operation"
+          schema:
+            type: "string"
+        409:
+          description: "conflict. another explorer found with same name"
+          schema:
+            type: "string"
+  /explorer/{explorername}:
+    get:
+      tags:
+      - "explorer"
+      summary: "Get explorer by explorer name"
+      description: ""
+      operationId: "getExplorerByName"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - name: "explorername"
+        in: "path"
+        description: "The name that needs to be fetched. Use explorer1 for testing. "
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/ExplorerSummary"
+        400:
+          description: "Invalid explorername supplied"
+        404:
```

Comment:
Dont have to specify 400 and 404 unless they serve a special purpose other than not found and bad request.

Reply:
Problem model
```
  Problem:
    type: object
    properties:
      error_code:
        type: string
        example: "NOT_FOUND"
      message:
        type: string
        example: "Error description"
```